k8s-install/scripts/install_cni_test.go

```diff
@@ -0,0 +1,204 @@
+				err := runCniContainer()
+				Expect(err).NotTo(HaveOccurred())
+
+				// Get a list of files in the defualt CNI bin location.
+				files, err := ioutil.ReadDir("tmp/bin")
+				if err != nil {
+					Fail("Could not list the files in tmp/bin")
+				}
+				names := []string{}
+				for _, file := range files {
+					names = append(names, file.Name())
+				}
+
+				// Get a list of files in the default location for CNI config.
+				files, err = ioutil.ReadDir("tmp/net.d")
+				if err != nil {
+					Fail("Could not list the files in tmp/net.d")
+				}
+				for _, file := range files {
+					names = append(names, file.Name())
+				}
+
+				Expect(names).To(ContainElement("calico"))
+				Expect(names).To(ContainElement("calico-ipam"))
+				Expect(names).To(ContainElement("10-calico.conf"))
+			})
+			It("Should parse and output a templated config", func() {
+				err := runCniContainer()
+				Expect(err).NotTo(HaveOccurred())
+
+				expectedFile := "expected_10-calico.conf"
+
+				var expected = []byte{}
+				if file, err := os.Open(expectedFile); err != nil {
+					Fail("Could not open " + expectedFile + " for reading")
+				} else {
+					_, err := file.Read(expected)
+					if err != nil {
+						Fail("Could not read from expected_10-calico.conf")
+					}
+					err = file.Close()
+					if err != nil {
+						fmt.Println("Failed to close expected_10-calico.conf")
+					}
+				}
+
+				var received = []byte{}
+				if file, err := os.Open("tmp/net.d/10-calico.conf"); err != nil {
+					Fail("Could not open 10-calico.conf for reading")
+				} else {
+					_, err := file.Read(received)
+					if err != nil {
+						Fail("Could not read from 10-calico.conf")
+					}
+					err = file.Close()
+					if err != nil {
+						fmt.Println("Failed to close 10-calico.conff")
+					}
+				}
+
+				Expect(expected).To(Equal(received))
+			})
+		})
+
+		Context("With modified env vars", func() {
+			It("Should rename '10-calico.conf' to '10-test.conflist'", func() {
+				err := runCniContainer("-e", "CNI_CONF_NAME=10-calico.conflist")
+				Expect(err).NotTo(HaveOccurred())
+
+				expectedFile := "expected_10-calico.conf"
+
+				var expected = []byte{}
+				if file, err := os.Open(expectedFile); err != nil {
+					Fail("Could not open " + expectedFile + " for reading")
+				} else {
+					_, err := file.Read(expected)
+					if err != nil {
+						Fail("Could not read from expected_10-calico.conf")
+					}
+					err = file.Close()
+					if err != nil {
+						fmt.Println("Failed to close expected_10-calico.conf")
+					}
+				}
+
+				var received = []byte{}
+				if file, err := os.Open("tmp/net.d/10-calico.conflist"); err != nil {
+					Fail("Could not open 10-calico.conflist for reading")
+				} else {
+					_, err := file.Read(received)
+					if err != nil {
+						Fail("Could not read from 10-calico.conflist.conf")
+					}
+					err = file.Close()
+					if err != nil {
+						fmt.Println("Failed to close 10-calico.conflist.conff")
```

Comment:
typo: 10-calico-conflist.conff should just be 10-calico.conflist (A few other places in this PR with the same issue)

Reply:
Gah, how did I manage to do that.